scope.go

```diff
@@ -482,6 +493,43 @@ func (s *scope) fullyQualifiedName(name string) string {
 	return fmt.Sprintf("%s%s%s", s.prefix, s.separator, name)
 }
 
+func (s *scope) uniqueID(name string, tags map[string]string) string {
+	name = s.fullyQualifiedName(name)
+	l := len(name)
+	for k, v := range tags {
+		l += len(k) + len(v)
+	}
+
+	// size the buffer to include separator bytes
+	buf := bytes.NewBuffer(make([]byte, 0, l+2*len(tags)+1))
```

Comment:
We could add a `Buffer` as a field of a `scope` and reuse it in calls to `Snapshot`. We would have to add a lock for it though to guard against concurrent calls to `uniqueID` from concurrent calls to `Snapshot`. I kind of like that approach.